Matrix.py

```diff
@@ -31,31 +22,39 @@ def __getitem__(self, item):
 
     def __repr__(self):
         """prints the matrix in a format that can be used for build another Matrix"""
-        return f"Matrix({self.data})"
+        return f"{self.__class__.__name__}({self.data})"
 
-    # str returns the Matrix in the mathemathic format
     def __str__(self):
         matrix_in_lines = ""
         for i in range(len(self.data)):
             matrix_in_lines += (str(self.data[i]) + "\n")
         return matrix_in_lines
 
-    # unity method creates a matrix of zeros with a diagonal of ones (unity is the Matrix's size)
     @staticmethod
-    def unity(unit):
-        major_list = [[1 if i == index else 0 for i in range(unit)] for index in range(unit)]
-        sub_list = [tuple(lt) for lt in major_list]
-        # converts list of lists to type Matrix. lt are the inside lists
-        matrix_to_return = Matrix((tuple(sub_list)))
-        return matrix_to_return
-
-    # ones method returns a Matrix full of ones which sizes determined by "unit"
+    def unity(size):
+        """ unity method creates a matrix of zeros with a diagonal of ones
+        args: size(int)- the size of the matrix and lines.
+        returns: a matrix full of 0 and 1's diagonal
+        exceptions: ValueError for non-ints"""
+        if type(size) is int:
+            major_list = [[1 if i == index else 0 for i in range(size)] for index in range(size)]
+            # converts list of lists to type Matrix. lt are the inside lists
```

Comment:
There really is no need for most of those docs.
You should add doc inside of a function, only if its really unclear what i means

Reply:
can i remove the docs from the add, sub and div?

Reply:
yes

Reply:
didn't remove it